### PR TITLE
Add at-functions for extractHash and extractValue

### PR DESCRIPTION
**Pull request by eth-r**
This enables further savings

Refs https://github.com/keep-network/tbtc-v2/issues/257

### CODE REVIEW

File: solidity/contracts/BTCUtils.sol [hunk 1]
```diff
@@ -534,46 +564,62 @@ library BTCUtils {
...
+    function extractHashAt(
+        bytes memory _output,
+        uint256 _at,
+        uint256 _len
+    ) internal pure returns (bytes memory) {
+        uint8 _scriptLen = uint8(_output[_at]);
 
         // don't have to worry about overflow here.
-        // if _scriptLen + 9 overflows, then output.length would have to be < 9
+        // if _scriptLen + 9 overflows, then output length would have to be < 9
```
Comment (pdyraga) [inline solidity/contracts/BTCUtils.sol]: Should it still be `_scriptLen + 9` or maybe `_scriptLen + 1` now?

**Comment by tomaszslabon** [comment]
Looks good to me.

**Review by tomaszslabon — APPROVED** [review]

File: solidity/contracts/BTCUtils.sol (hunk 1 above)
Comment (lukasz-zimnoch) [reply solidity/contracts/BTCUtils.sol]: https://github.com/keep-network/bitcoin-spv/pull/7/commits/5708f4c37de95088658957949e2faa86fa38c463

Comment (eth-r) [reply solidity/contracts/BTCUtils.sol]: Yeah, it should be `_scriptLen + 1`

Comment (pdyraga) [reply solidity/contracts/BTCUtils.sol]: What about the rest of this comment where we still mention  < 9?

**Pull request merged**